appyx-components/backstack/common/src/commonMain/kotlin/com/bumble/appyx/components/backstack/ui/parallax/TargetUiState.kt

```diff
@@ -0,0 +1,31 @@
+package com.bumble.appyx.components.backstack.ui.parallax
+
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+import com.bumble.appyx.interactions.core.ui.context.UiContext
+import com.bumble.appyx.interactions.core.ui.property.impl.Position
+import com.bumble.appyx.interactions.core.ui.state.MutableUiStateSpecs
+
+@MutableUiStateSpecs
+class TargetUiState(
+    val position: Position.Target = Position.Target(DpOffset.Zero),
+) {
+
+    constructor(
+        elementWidth: Float,
+        offsetMultiplier: Float,
+    ) : this(
+        position = Position.Target(
+            DpOffset(
+                x = (offsetMultiplier.coerceIn(-0.5f, 1.5f) * elementWidth).dp,
+                y = 0.dp
+            )
+        ),
+    )
+
+    fun toMutableState(uiContext: UiContext): MutableUiState =
+        MutableUiState(
+            uiContext = uiContext,
+            position = Position(uiContext, position),
+        )
```

Comment:
This straight forward implementation is already provided by `MutableStateSpecs` auto-generated code. My suggestion is to remove it.